node/gum/proc-macro/src/lib.rs

```diff
@@ -43,6 +43,27 @@ pub fn warn(item: proc_macro::TokenStream) -> proc_macro::TokenStream {
 	gum(item, Level::Warn)
 }
 
+/// Print a warning or debug level message depending on their frequency
+#[proc_macro]
+pub fn warn_if_frequent(item: proc_macro::TokenStream) -> proc_macro::TokenStream {
+	let ArgsIfFrequent { freq, max_rate, rest } = parse2(item.into()).unwrap();
+
+	let freq_expr = freq.expr;
+	let max_rate_expr = max_rate.expr;
+	let debug: proc_macro2::TokenStream = gum(rest.clone().into(), Level::Debug).into();
+	let warn: proc_macro2::TokenStream = gum(rest.into(), Level::Warn).into();
+
+	let stream = quote! {
+		if #freq_expr .is_frequent(#max_rate_expr) {
```

Comment:
I'm wondering how much overhead is added by the EMA stuff. Can we make this not use floats and be faster ?